aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1297,20 +1325,34 @@ AT_ERROR("svd: MAGMA library not found in "
   scalar_t wkopt;
   magma_int_t* iwork;
   ALLOCATE_ARRAY(iwork, magma_int_t, 8 * k);
-  magmaSvd<scalar_t>(jobz, m, n, self_data, m, S_data, U_data, m, VT_data, n, &wkopt, lwork, iwork, &info);
-  lwork = magma_int_cast(wkopt, "work_size");
+
+  value_t* rwork = nullptr;
+  if (isComplexType(at::typeMetaToScalarType(self.dtype()))) {
+    auto mx = std::max(m, n);
+    int64_t lrwork;
+    if (jobchar == 'N'){
+      lrwork = 7 * k;
```

Comment:
I think it's a good idea to rename the `k` variable to `mn` to match CPU code and MAGMA documentation.

Reply:
LAPACK <= 3.6 had a bug requiring `7 * k`. CPU code has `lrwork = 7 * k;` to be compatible with older LAPACK versions. I believe this is not required for MAGMA anymore and the more optimal value is `lrwork = 5 * k;`